core/Names.cc

```diff
@@ -14,7 +14,16 @@ using namespace std;
 
 namespace sorbet::core {
 
-NameRef::NameRef(const GlobalState &gs, u4 id) : DebugOnlyCheck(gs, id), _id{id} {}
+NameRef::NameRef(const GlobalState &gs, NameKind kind, u4 id)
+    : DebugOnlyCheck(gs, id), _id{id | (static_cast<u4>(kind) << ID_BITS)} {
+    // If this fails, the symbol table is too big :(
+    ENFORCE_NO_TIMER(id <= ID_MASK);
```

Comment:
Same questions as in `NameRef.h` about not masking the id.

Reply:
Done